cloudnew/ec2_util.go

```diff
@@ -0,0 +1,142 @@
+package cloudnew
+
+import (
+	"math"
+	"os"
+	"os/user"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/pkg/errors"
+)
+
+const (
+	HostExpireDays      = 10
+	NameTimeFormat      = "20060102150405"
+	SpawnHostExpireDays = 30
+)
+
+func makeBlockDeviceMappings(mounts []cloud.MountPoint) ([]*ec2.BlockDeviceMapping, error) {
+	mappings := []*ec2.BlockDeviceMapping{}
+	for i, mount := range mounts {
+		if mount.DeviceName == "" {
+			return nil, errors.Errorf("missing 'device_name': %+v", mount)
+		}
+		if mount.VirtualName == "" {
+			return nil, errors.Errorf("missing 'virtual_name': %+v", mount)
+		}
+		mappings = append(mappings, &ec2.BlockDeviceMapping{
+			DeviceName:  &mounts[i].DeviceName,
+			VirtualName: &mounts[i].VirtualName,
+		})
+	}
+	return mappings, nil
+}
+
+//makeTags populates a map of tags based on a host object, which contain keys
+//for the user, owner, hostname, and if it's a spawnhost or not.
+func makeTags(intentHost *host.Host) map[string]string {
+	// get requester host name
+	hostname, err := os.Hostname()
+	if err != nil {
+		hostname = "unknown"
+	}
+
+	// get requester user name
+	var username string
+	user, err := user.Current()
```

Comment:
this is always going to be "evergreen" right? I guess it makes local testing more clear